tracee-ebpf/tracee/proc.go

```diff
@@ -0,0 +1,87 @@
+package tracee
+
+import (
+	"github.com/prometheus/procfs"
+)
+
+func gatherEntireProcessTree() (map[uint32][]uint32, error) {
+	fs, err := procfs.NewFS("/proc")
+	if err != nil {
+		return nil, err
+	}
+
+	procs, err := fs.AllProcs()
+	if err != nil {
+		return nil, err
+	}
+
+	processMap := map[uint32][]uint32{}
+	for i := range procs {
+		stat, err := procs[i].Stat()
+		if err != nil {
+			return nil, err
+		}
+		processMap[uint32(stat.PID)] = processMap[uint32(stat.PID)]
```

Comment:
What does this line do?

Reply:
This is so that all PIDs are recorded in the map, even if they don't have any child processes.

Reply:
But this is a self-assignment of processMap[uint32(stat.PID)] to itself, how does it achieves this?

Reply:
Lol that's actually interesting, I never stopped to think that it shouldn't be valid code. It does work though.

Played with it here:
 https://play.golang.org/p/2uHNoyTcnfh

Reply:
It seems that it doesn't work:
./prog.go:30:2: self-assignment of x[foo] to x[foo]

Reply:
That's a go vet error, but i'm still not entirely sure why it doesn't complain in this context.